[EV-6333] Clarify flow log policy fields in datatypes reference (#2472)
tigera/docs: calico-cloud/observability/elastic/flow/datatypes.mdx

| `all_policies`                    | array of keywords | **Deprecated.** This field is a legacy field calculated at flow start time that combines both enforced and staged policies. Because it mixes actual verdicts with simulated ones, it can be misleading and is planned for removal. <br/><br/>**Note:** This field may still appear in logs from older clusters reporting to a newer management cluster. |
| `enforced_policies`               | array of keywords | The policies that were actually enforced on the flow. This list is determined by the dataplane when the flow starts and remains fixed for the flow's lifetime. It represents the concrete actions taken on the traffic. |
| `pending_policies`                | array of keywords | A simulation of what policy evaluation would look like at the time of log generation. It represents a hypothetical restart of the flow where all staged policies are considered active. <br/>This field captures two types of updates: <br/> 1. Changes to active policies made after the flow started. <br/> 2. Staged policies treated as if they were enforced. |

> | `transit_policies`                | array of keywords | Policies applied to traffic transiting through the node (Apply On Forward) or during pre-DNAT processing. This field functions like `enforced_policies` but specifically for traffic being forwarded rather than terminating at a local workload. |

Reviewer: ```suggestion
| `transit_policies`                | array of keywords | Policies applied to traffic transiting through the node (Apply On Forward) or during pre-DNAT processing. This field functions like `enforced_policies` but specifically for traffic being evaluated at the host endpoint. |
```

Would this be better? I was not sure about "local workload".

Reviewer 2: I'm not sure myself what the correct options is. But 'the host endpoint' is much clearer that 'a local workload' in this line.
